crates/api/src/trap.rs

```diff
@@ -62,22 +71,45 @@ impl fmt::Display for Trap {
 impl std::error::Error for Trap {}
 
 #[derive(Debug)]
-pub struct FrameInfo;
+pub struct FrameInfo {
+    module_name: Option<String>,
+    func_index: u32,
+}
 
 impl FrameInfo {
     pub fn instance(&self) -> *const Instance {
         unimplemented!("FrameInfo::instance");
     }
 
-    pub fn func_index() -> usize {
-        unimplemented!("FrameInfo::func_index");
+    pub fn func_index(&self) -> usize {
+        self.func_index as usize
     }
 
-    pub fn func_offset() -> usize {
+    pub fn func_offset(&self) -> usize {
         unimplemented!("FrameInfo::func_offset");
     }
 
-    pub fn module_offset() -> usize {
+    pub fn module_offset(&self) -> usize {
         unimplemented!("FrameInfo::module_offset");
     }
+
+    pub fn module_name(&self) -> Option<&String> {
+        self.module_name.as_ref()
+    }
+}
+
+impl From<BacktraceFrame> for FrameInfo {
```

Comment:
Could this be a `pub(crate)` function instead of a `From` impl to avoid adding `wasmtime_runtime` to the public API?